src/classes/RD_RecurringDonations.cls

```diff
@@ -1214,4 +1543,92 @@ public class RD_RecurringDonations {
         }
         return calcDate;
     }
+
+    /*******************************************************************************************************
+    * @description Returns true if the RD changes should trigger recalculation of the child opportunities
+    * @param new The updated Recurring Donation
+    * @param old The original Recurring Donation before updates
+    * @return boolean
+    */
+    @testVisible
+    private static Boolean needsRecalculation(npe03__Recurring_Donation__c newRD, npe03__Recurring_Donation__c oldRD) {
+
+        if(newRD.npe03__Organization__c != oldRD.npe03__Organization__c
+                || newRD.npe03__Contact__c != oldRD.npe03__Contact__c
+                || newRD.npe03__Amount__c != oldRD.npe03__Amount__c
+                || newRD.npe03__Installment_Period__c != oldRD.npe03__Installment_Period__c
+                || newRD.npe03__Installments__c != oldRD.npe03__Installments__c
+                || newRD.Always_Use_Last_Day_Of_Month__c != oldRD.Always_Use_Last_Day_Of_Month__c
+                || newRD.npe03__Schedule_Type__c != oldRD.npe03__Schedule_Type__c
+                || newRD.npe03__Open_Ended_Status__c != oldRD.npe03__Open_Ended_Status__c
+                || newRD.npe03__Next_Payment_Date__c != oldRD.npe03__Next_Payment_Date__c
+                || newRD.npe03__Recurring_Donation_Campaign__c != oldRD.npe03__Recurring_Donation_Campaign__c
+                ) {
+            return true;
+        }
+
+        for (String s : customFieldMappingSettings.keySet()) {
+            npe03__Custom_Field_Mapping__c cfm = customFieldMappingSettings.get(s);
+            if(newRD.get(cfm.npe03__Recurring_Donation_Field__c) != oldRD.get(cfm.npe03__Recurring_Donation_Field__c)) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    /*******************************************************************************************************
+    * @description Creates a new Opportunity on a given date for a given RD
+    * @param oppCloseDate The close date for the new Opportunity
+    * @param r The parent Recurring Donation
+    * @param rdcounter Opportunity counter for naming and Recurring_Donation_Installment_Number__c
+    * @return Opportunity
+    */
+    private static Opportunity newOpportunityOnDate(Date oppCloseDate, npe03__Recurring_Donation__c r, Integer rdcounter) {
+
+        npe03__Recurring_Donations_Settings__c rds = UTIL_CustomSettingsFacade.getRecurringDonationsSettings();
+
+        Opportunity o = new Opportunity();
+        o.CloseDate = oppCloseDate;
+        o.OwnerId = r.OwnerId;
+
+
+        if(isMultiCurrency){
+            //set the donation currency field to equal the recurring donation currency
+            o.put(OppCurrencyField,r.get(RDCurrencyField));
+        }
+
+        o.Amount = r.npe03__Amount__c;
+
+        String oName = '';
```

Comment:
is this method only used for Open RD's or Fixed Length too?  If also for Fixed Length, then the naming strategy needs to change.

Reply:
fixed in latest commit. good catch.